tools/mpremote/mpremote/main.py

```diff
@@ -166,7 +167,12 @@ def do_connect(args):
             for p in sorted(serial.tools.list_ports.comports()):
                 print(
                     "{} {} {:04x}:{:04x} {} {}".format(
-                        p.device, p.serial_number, p.vid, p.pid, p.manufacturer, p.product
+                        p.device,
+                        p.serial_number,
+                        p.pid if isinstance(p.pid, int) else 0,
+                        p.vid if isinstance(p.vid, int) else 0,
```

Comment:
You seem to have accidentally swapped the order of vid and pid?

Reply:
Gaah!  I certainly tested this but somehow it got swapped... now fixed, thanks for picking it up!